routes/ecom/modules/list-payments.js

```diff
@@ -3,12 +3,43 @@
 // generate default payment gateway object
 const newPaymentGateway = require(process.cwd() + '/lib/new-payment-gateway')
 
+const checkDiscountItems = (paymentGateway, items) => {
+  const discount = paymentGateway.discount
+  const productIds = discount && discount.product_ids
+
+  if (!productIds || !productIds.length) return
+
+  if (items && items.length) {
+    if (discount.apply_at === 'subtotal' && discount.type === 'percentage' && discount.value > 0) {
+      let discountValue = 0
+      items.forEach(item => {
+        if (productIds.includes(item.product_id)) {
+          discountValue += (item.quantity) * (item.final_price || item.price)
+        }
+      })
+      if (discountValue) {
+        discountValue *= discount.value / 100
+        const newDiscount = {
+          apply_at: discount.apply_at,
+          value: discountValue,
+          type: 'fixed'
+        }
+        paymentGateway.discount = newDiscount
+      }
+    }
+  } else if (productIds && productIds.length) {
+    // because the discount is not valid for all products and it is not good to use this discount in the showcase
+    delete paymentGateway.discount
+  }
```

Comment:
```suggestion
  }
  delete paymentGateway.discount
```

Não chegaria aqui sem `productIds.length`

Reply:
E aqui acho que faria sentido manter o desconto (sem alterar o valor) se todos os itens do carrinho atual baterem com os `productIds`, ou seja, se o desconto não for percentual e/ou não for no subtotal ele ainda seria aplicado ("as is") se o carrinho conter apenas produtos selecionados